DashWallet/Sources/UI/Payments/Amount/Model/BaseAmountModel.swift

```diff
@@ -49,10 +50,8 @@ class BaseAmountModel {
 
     var mainAmount: AmountObject!
     var supplementaryAmount: AmountObject!
-    var amount: AmountObject {
-        activeAmountType == .main ? mainAmount : supplementaryAmount
-    }
-
+    @Published var amount: AmountObject!
+    
```

Comment:
Currently, with how the `BaseAmountViewController` works, it's confusing where descendants should watch for amount updates - in an overridden method or in the `amountChangeHandler` of the model.

To fix this, I've hidden the methods and only the `amount` field is exposed on the model. It's made `@Published` to allow multiple subscribers with Combine.